rest/data/test_results_test.go

```diff
@@ -1 +1,220 @@
+	conf := dbModel.NewCedarConfig(s.env)
+	conf.Bucket = dbModel.BucketConfig{TestResultsBucket: s.tempDir}
+	s.Require().NoError(conf.Save())
+
+	testResultInfos := []dbModel.TestResultsInfo{
+		{
+			Project:     "test",
+			Version:     "0",
+			Variant:     "linux",
+			TaskName:    "task0",
+			TaskID:      "task1",
+			Execution:   0,
+			RequestType: "requesttype",
+			Mainline:    true,
+		},
+		{
+			Project:     "test",
+			Version:     "0",
+			Variant:     "linux",
+			TaskName:    "task0",
+			TaskID:      "task1",
+			Execution:   1,
+			RequestType: "requesttype",
+			Mainline:    true,
+		},
+		{
+			Project:     "test",
+			Version:     "0",
+			Variant:     "linux",
+			TaskName:    "task2",
+			TaskID:      "task3",
+			Execution:   0,
+			RequestType: "requesttype",
+			Mainline:    true,
+		},
+	}
+
+	for _, testResultsInfo := range testResultInfos {
+		testResults := dbModel.CreateTestResults(testResultsInfo, dbModel.PailLocal)
+
+		opts := pail.LocalOptions{
+			Path:   s.tempDir,
+			Prefix: filepath.Join("test_results", testResults.Artifact.Prefix),
+		}
+		bucket, err := pail.NewLocalBucket(opts)
+		s.Require().NoError(err)
+
+		testResults.Setup(s.env)
+		err = testResults.SaveNew(s.ctx)
+
+		s.Require().NoError(err)
+		s.testResults[testResults.ID] = *testResults
+
+		for i := 0; i < 3; i++ {
+			result := dbModel.TestResult{
+				TaskID:         testResults.Info.TaskID,
+				Execution:      testResults.Info.Execution,
+				TestName:       fmt.Sprintf("test%d", i),
+				Trial:          0,
+				Status:         "teststatus",
+				LogURL:         "testurl",
+				LineNum:        0,
+				TaskCreateTime: time.Now().Add(-3 * time.Second),
+				TestStartTime:  time.Now().Add(-2 * time.Second),
+				TestEndTime:    time.Now().Add(-1 * time.Second),
+			}
+
+			data, err := bson.Marshal(result)
+			s.Require().NoError(err)
+			s.Require().NoError(bucket.Put(s.ctx, result.TestName, bytes.NewReader(data)))
+		}
+	}
+}
+
+func (s *testResultsConnectorSuite) TearDownSuite() {
+	defer s.cancel()
+	s.NoError(os.RemoveAll(s.tempDir))
+	s.NoError(s.env.GetDB().Drop(s.ctx))
+}
+
+func (s *testResultsConnectorSuite) TestFindTestResultByTestNameExists() {
+	optsList := []TestResultsOptions{{
+		TaskID:    "task1",
+		Execution: 1,
+		TestName:  "test1",
+	}, {
+		TaskID:         "task1",
+		EmptyExecution: true,
+		TestName:       "test1",
+	}}
+	for _, opts := range optsList {
+		testResults := dbModel.TestResults{}
+		testResults.Setup(s.env)
+		findOpts := dbModel.TestResultsFindOptions{
+			TaskID:         opts.TaskID,
+			Execution:      opts.Execution,
+			EmptyExecution: opts.EmptyExecution,
+		}
+		s.Require().NoError(testResults.FindByTaskID(s.ctx, findOpts))
+		bucket, err := testResults.GetBucket(s.ctx)
+		s.Require().NoError(err)
+
+		tr, err := bucket.Get(s.ctx, opts.TestName)
+		s.Require().NoError(err)
+		defer func() {
+			closeErr := tr.Close()
+			grip.Warning(errors.Wrap(closeErr, "some message"))
```

Comment:
Since this is a test, we don't need to use grip, you can just change this to `s.NoError(tr.Close())`